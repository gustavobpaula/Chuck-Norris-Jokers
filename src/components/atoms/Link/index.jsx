import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const LinkTo = ({ title, to, children }) => (
  <Link to={to} title={title}>
    {children}
  </Link>
);

LinkTo.propTypes = {
  children: PropTypes.node.isRequired,
  title: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired,
};

export default LinkTo;
