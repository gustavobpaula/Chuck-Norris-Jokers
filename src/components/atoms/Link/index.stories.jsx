import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import Link from './index';

export default {
  component: Link,
  title: 'Atoms/Link',
};

export const element = () => (
  <BrowserRouter>
    <Link title="Link Component" to="/">
      Component Link
    </Link>
  </BrowserRouter>
);

element.story = {
  name: 'Default',
};
