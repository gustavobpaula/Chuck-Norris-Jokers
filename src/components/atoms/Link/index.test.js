import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import Link from './index';

describe('<Link />', () => {
  it('should mount properly', async () => {
    const wrapper = mount(
      <BrowserRouter>
        <Link title="Link Component" to="/">
          Component Link
        </Link>
      </BrowserRouter>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
