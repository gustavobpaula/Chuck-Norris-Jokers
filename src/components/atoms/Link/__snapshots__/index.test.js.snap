// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> should mount properly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <LinkTo
      className="link"
      title="Link Component"
      to="/"
    >
      <Link
        className="link"
        title="Link Component"
        to="/"
      >
        <LinkAnchor
          className="link"
          href="/"
          navigate={[Function]}
          title="Link Component"
        >
          <a
            className="link"
            href="/"
            onClick={[Function]}
            title="Link Component"
          >
            Component Link
          </a>
        </LinkAnchor>
      </Link>
    </LinkTo>
  </Router>
</BrowserRouter>
`;
